src/expr/src/scalar/mod.rs

```diff
@@ -632,6 +632,23 @@ impl MirScalarExpr {
                                     e.typ(&relation_type).scalar_type,
                                 ),
                             };
+                        } else if *func == VariadicFunc::ListIndex && exprs[1].is_literal_ok() {
+                            let literal = match exprs[1].as_literal().unwrap().unwrap() {
+                                Datum::Int64(data) => usize::try_from(data - 1).ok(),
+                                _ => None,
+                            };
+                            if let Some(index_literal) = literal {
+                                if let MirScalarExpr::CallVariadic {
+                                    func,
```

Comment:
You should be able to change this line to `func: VariadicFunc::ListCreate {..},` and skip the matching for `VariadicFunc::ListCreate { .. }` further down.